pkg/bot/lark.go

```diff
@@ -0,0 +1,231 @@
+	//P2pChatCreate eventType When a session is first created with the bot
+	P2pChatCreate = "p2p_chat_create"
+	//AddUserToChat eventType When a user is added to a chat group
+	AddUserToChat = "add_user_to_chat"
+	//LarkStartMsg lark start message
+	LarkStartMsg = "Hello from botkube~ Play with me by at botkube <commands>"
+)
+
+// LarkBot listens for user's message, execute commands and sends back the response
+type LarkBot struct {
+	AllowKubectl     bool
+	RestrictAccess   bool
+	ClusterName      string
+	DefaultNamespace string
+	Port             int
+	MessagePath      string
+	LarkClient       *utils.LarkClient
+}
+
+// NewLarkBot returns new Bot object
+func NewLarkBot(c *config.Config) Bot {
+	larkConf := c.Communications.Lark
+	appSettings := core.NewInternalAppSettings(core.SetAppCredentials(larkConf.AppID, larkConf.AppSecret),
+		core.SetAppEventKey(larkConf.VerificationToken, larkConf.EncryptKey))
+	conf := core.NewConfig(core.Domain(larkConf.Endpoint), appSettings, core.SetLoggerLevel(core.LoggerLevelError))
+	return &LarkBot{
+		AllowKubectl:     c.Settings.Kubectl.Enabled,
+		RestrictAccess:   c.Settings.Kubectl.RestrictAccess,
+		ClusterName:      c.Settings.ClusterName,
+		DefaultNamespace: c.Settings.Kubectl.DefaultNamespace,
+		Port:             c.Communications.Lark.Port,
+		MessagePath:      c.Communications.Lark.MessagePath,
+		LarkClient:       utils.NewLarkClient(conf),
+	}
+}
+
+// Execute commands sent by users
+func (l *LarkBot) Execute(e map[string]interface{}) {
+	event, err := accessAndTypeCastToMap(LarkEvent, e)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToMap: %s", err.Error())
+		return
+	}
+
+	chatType, err := accessAndTypeCastToString(ChatType, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+
+	text, err := accessAndTypeCastToString(Text, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+
+	executor := execute.NewDefaultExecutor(text, l.AllowKubectl, l.RestrictAccess, l.DefaultNamespace,
+		l.ClusterName, config.LarkBot, "", true)
+	response := executor.Execute()
+
+	if chatType == Group {
+		chatID, err := accessAndTypeCastToString(OpenChatID, event)
+		if err != nil {
+			log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+			return
+		}
+
+		l.LarkClient.SendTextMessage(ChatID, chatID, response)
+		return
+	}
+	openID, err := accessAndTypeCastToString(OpenID, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+	l.LarkClient.SendTextMessage(OpenID, openID, response)
+}
+
+// Start starts the lark server and listens for lark messages
+func (l *LarkBot) Start() {
+	// See: https://open.larksuite.com/document/ukTMukTMukTM/ukjNxYjL5YTM24SO2EjN
+	// message
+	larkConf := l.LarkClient.Conf
+	event.SetTypeCallback(larkConf, Message, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
```

Comment:
Do we need to log this? Should we move it to debug level? Also, let's add an additional message if required.